lib/methods/dynamics/webApi/create.js

```diff
@@ -17,49 +18,47 @@ module.exports = (
   } = internals.dynamics
 
   /**
-   * Creates an enrolment between a service and a contact, for the passed organisation and role
-   *
-   * @param {string} lobServiceRoleId dynamics line of business service role guid
    * @param {string} contactId dynamics contact guid
-   * @param {string} organisationAccountId dynamics organisation account guid
    * @param {string} connectionDetailsId dynamics connection detail guid
    * @param {string} enrolmentStatus dynamics enrolment status id - mappings.enrolmentStatus
-   * @param {string} enrolmentType dynamics enrolment type string - mappings.enrolmentType
-   * @return {Promise<T | Object>}
+   * @param {string} [organisationAccountId] dynamics organisation account guid
+   * @param {string} [lobServiceId] dynamics service guid
+   * @param {string} [lobServiceRoleId] dynamics line of business service role guid
+   * @param {boolean} [verified=false] boolean indicating whether the connection has been verified or not
+   * @returns {Promise<Object>}
    */
-  const createEnrolment = async (lobServiceRoleId, contactId, organisationAccountId, connectionDetailsId, enrolmentStatus, enrolmentType) => {
-    const request = await createEnrolment.buildRequest(lobServiceRoleId, contactId, organisationAccountId, connectionDetailsId, enrolmentStatus, enrolmentType)
+  const createEnrolment = async (contactId, connectionDetailsId, enrolmentStatus, organisationAccountId = undefined, lobServiceId = undefined, lobServiceRoleId = undefined, verified = false) => {
```

Comment:
Signature could be made to use a model that with validation (Non blocking review)

Reply:
Interested in this, what do you mean?

Reply:
To speed up understand and therefore (hopefully) development we could pass in a context/model object that can be named to describe what it is to others and can be validated against instead of having to puzzle through all the params and collect them. Maybe that model object could be produced from another function primed with params automatically.

Reply:
This comment is a global strategy to the application and not specific to the apps which is why it doesn't block the review.